webpack/scenes/ContentViews/components/EnvironmentPaths/EnvironmentPaths.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { STATUS } from 'foremanReact/constants';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { Form, FormGroup, Checkbox } from '@patternfly/react-core';
+import { selectEnvironmentPaths, selectEnvironmentPathsStatus } from './EnvironmentPathSelectors';
+import EnvironmentLabels from '../EnvironmentLabels';
+import './EnvironmentPaths.scss';
+import Loading from '../../../../components/Loading';
+
+const EnvironmentPaths = ({ userCheckedItems, setUserCheckedItems }) => {
+  const environmentPathResponse = useSelector(selectEnvironmentPaths);
+  const environmentPathStatus = useSelector(selectEnvironmentPathsStatus);
+  const environmentPathLoading = environmentPathStatus === STATUS.PENDING;
+
+  const oncheckedChange = (checked, env) => {
+    if (checked) {
+      setUserCheckedItems([...userCheckedItems, env]);
+    } else {
+      setUserCheckedItems(userCheckedItems.filter(item => item.id !== env.id));
+    }
+  };
+
+  const buildformGroups = (results) => {
```

Comment:
I don't think this should be a function.  This is part of render.

1. Change the `results.forEach` to a `results.map`
3. Don't reassign with `concat` each time - Have each item in the `environments.map` return a React.Fragment with the `<FormGroup>` and the `<hr>`
4. Move it all inside the `<Form>...</Form>` on line 66.